### PR TITLE
Initial Lightning implementation

### PR DESCRIPTION
**Pull request by dangeross**
This PR adds the initial lightning implementation to the BitBoxApp using the Breez SDK. Two vendor libraries are added to support this PR [breez-sdk-go](https://github.com/digitalbitbox/bitbox-wallet-app/pull/github.com/breez/breez-sdk-go) and [go-bip39](https://github.com/digitalbitbox/bitbox-wallet-app/pull/github.com/tyler-smith/go-bip39). The application successfully builds (after manually adding the mock entropy and API key) for Android, macOS, and Windows (manual build process).

It currently supports:

Creation of a node from a mock deterministic entropy
Deactivation of a node
Display of node balance and payment history in the Lightning view
Send and receive BOLT11 payments
See commit messages for details.

### CODE REVIEW

**Review by Beerosagos — COMMENTED** [review]
Thanks @dangeross for the great job! 
I left a few first comments about the PR here and there, but I think that it would be more efficient to merge it as-is and address all the changes in future PRs, so that we can work on things in parallel. More comments will probably come later on, even after merging the PR. 

cc @thisconnect @benma do you think we can proceed with merging or is there something we should absolutely change before that?

**Comment by Beerosagos** [comment]
I'm not sure if I like the request/response DTOs and functions in the backend for objects that are just used by a single handler.
That is quite different from how we manage that kind of stuff in the rest of the backend and I feel like it would be better to define objects and useful functions inside the handler itself, if they are not needed anywhere else. 
No strong opinion, tho. cc @benma

**Comment by thisconnect** [comment]
> do you think we can proceed with merging or is there something we should absolutely change before that?

my biggest concern is that we miss something wrong int he merge commits.

@dangeross did you have any noticeable merge conflicts, something we should take a close look?

File: frontends/web/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+{
+  "endOfLine": "lf",
+  "printWidth": 140,
+  "semi": true,
+  "singleQuote": true,
+  "tabWidth": 2,
+  "trailingComma": "none"
+}
```
Comment (thisconnect) [inline frontends/web/.prettierrc:9]: we don't use prettier at the moment.

File: frontends/web/yarn.lock
Comment (thisconnect) [inline frontends/web/yarn.lock:1]: we don't use yarn

File: vendor/github.com/breez/breez-sdk-go/breez_sdk/lib/darwin-amd64/libbreez_sdk_bindings.dylib
Comment (thisconnect) [inline vendor/github.com/breez/breez-sdk-go/breez_sdk/lib/darwin-amd64/libbreez_sdk_bindings.dylib:1]: I was confused about all these files, but IIRC android isn't linked... but why is there darwin and linux?
cc @benma

File: frontends/web/src/api/lightning.ts [hunk 2]
```diff
@@ -0,0 +1,961 @@
...
+};
+
+/**
+ * Returns a function that subscribes a callback on a "lightning/node-state"
+ * event to receive the latest state of the lightning node.
+ * Meant to be used with `useSubscribe`.
+ */
+export const subscribeNodeState = (cb: TSubscriptionCallback<NodeState>) => {
+  return subscribeEndpoint('lightning/node-info', cb);
+};
```
Comment (thisconnect) [inline frontends/web/src/api/lightning.ts:961]: This file is crazy. Was this generated somehow, if yes how?

Also there seems to be un-used stuff, i.e. EnvironmentType or ReverseSwapFeesRequest

**Review by thisconnect — COMMENTED** [review]
Checked frontends/web mostly for merge conflicts 👍

File: vendor/github.com/breez/breez-sdk-go/breez_sdk/lib/darwin-amd64/libbreez_sdk_bindings.dylib
Comment (Beerosagos) [reply vendor/github.com/breez/breez-sdk-go/breez_sdk/lib/darwin-amd64/libbreez_sdk_bindings.dylib:1]: I think this is just the stuff that goes inside the go breez dep. The lib supports android, we just didn't link it correctly yet. We should do something like @benma did here for windows https://github.com/digitalbitbox/bitbox-wallet-app/pull/2377/files

**Review by thisconnect — DISMISSED** [review]
frontend/web (b49a8b690cabbe72c825c7aad67ba559bf88c93c) looks good to me to merge, with some nit's but we can fix them on staging.

File: vendor/github.com/breez/breez-sdk-go/breez_sdk/lib/darwin-amd64/libbreez_sdk_bindings.dylib
Comment (dangeross) [reply vendor/github.com/breez/breez-sdk-go/breez_sdk/lib/darwin-amd64/libbreez_sdk_bindings.dylib:1]: The darwin, windows dll's and linux libraries are for the respective builds. They are already configured (in staging-ln) to be used in the makefiles:

https://github.com/digitalbitbox/bitbox-wallet-app/blob/0caffef896691489e1b612e101403cee8a1eaedd/frontends/qt/Makefile#L34

https://github.com/digitalbitbox/bitbox-wallet-app/blob/0caffef896691489e1b612e101403cee8a1eaedd/frontends/qt/Makefile#L57

https://github.com/digitalbitbox/bitbox-wallet-app/blob/0caffef896691489e1b612e101403cee8a1eaedd/frontends/qt/make_windows.sh#L23

File: frontends/web/src/api/lightning.ts (hunk 2 above)
Comment (dangeross) [reply frontends/web/src/api/lightning.ts:961]: This code is generated from rust for our react native package.  Currently it was just copy/pastes. Let me know if you want me to remove any unused types.

Comment (thisconnect) [reply frontends/web/src/api/lightning.ts:961]: It is at the moment unlcear how to deal with this file.

We could also remove the unused types / api calls later on staging. 

But if this is a generated file, we probably shouldn't do manual changes?

are you planing to re-generate this again if the rust package changes?

**Review by Beerosagos — APPROVED** [review]

File: frontends/web/src/routes/lightning/lightning.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            }
+          ></Header>
+          <div className="innerContainer scrollableContainer">
+            <div className="content padded">
+              <div className="flex flex-column flex-reverse-mobile">
+                <label className="labelXLarge flex-self-start-mobile hide-on-small">{t('accountSummary.availableBalance')}</label>
+                <div className="flex flex-row flex-between flex-item-center flex-column-mobile flex-reverse-mobile">
+                  <Balance balance={balance} />
+                  <label className="labelXLarge flex-self-start-mobile show-on-small">{t('accountSummary.availableBalance')}</label>
+                  {!isAccountEmpty && <ActionButtons {...actionButtonsProps} />}
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/routes/lightning/lightning.tsx:130]: Opened a PR https://github.com/digitalbitbox/bitbox-wallet-app/pull/2556

File: frontends/web/src/app.tsx [hunk 4]
```diff
@@ -47,13 +47,15 @@ import { WCWeb3WalletProvider } from './contexts/WCWeb3WalletProvider';
...
 import { sortAccounts } from './routes/account/utils';
+import { getLightningConfig, subscribeLightningConfig } from './api/lightning';
 
 export const App = () => {
   const { t } = useTranslation();
   const navigate = useNavigate();
 
   const accounts = useDefault(useSync(getAccounts, syncAccountsList), []);
   const devices = useDefault(useSync(getDeviceList, syncDeviceList), {});
+  const lightningConfig = useDefault(useSync(getLightningConfig, subscribeLightningConfig), { inactive: true });
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/app.tsx:58]: Opened a PR to move this into a context, https://github.com/digitalbitbox/bitbox-wallet-app/pull/2561

in the future we could move maybe other subscriptions into that context as well.

**Pull request merged**